library/macos/FLETextInputModel.m

```diff
@@ -70,9 +74,21 @@ static long GetExtentForRange(NSRange range) {
 
 @implementation FLETextInputModel
 
-- (instancetype)init {
+- (instancetype)initWithClientID:(nonnull NSNumber *)clientID
+                   configuration:(nonnull NSDictionary *)config {
   self = [super init];
   if (self != nil) {
+    _clientID = clientID;
+    _inputAction = config[kTextInputAction];
+    // There's more information that can be used from this dictionary.
+    // Add more as needed.
+    NSDictionary *inputTypeInfo = config[kTextInputType];
+    _inputType = inputTypeInfo[kTextInputTypeName];
+    if (!_clientID || !_inputAction || !_inputType) {
+      NSLog(@"Missing arguments for %@ init.", [self class]);
+      return nil;
```

Comment:
A `nonnull` return value annotation means "I guarantee that this method will not return nil". You can't both annotate it `nonnull` and use nil for error cases. If you want it to be nonnull, then you'll need to handle these with fallbacks, otherwise you need to make it `nullable`.

Reply:
Sorry, super silly mistake. Making it nullable since providing default clientIDs or inputTypes might cause other issues if we decide to keep track of state between models